src/lib.rs

```diff
@@ -94,3 +93,188 @@ mod tests {
         assert_eq!(p, expected);
     }
 }
+
+/// A safe wrapper for a path with only a single component.
+/// This prevents path traversal attacks.
+///
+/// It just allows a single normal path element and no parent, root directory or prefix like `C:`.
+#[derive(Clone, Debug, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct PathComponent {
+    path: PathBuf,
+}
+
+impl PathComponent {
+    /// It creates the wrapped `PathComponent` if it's valid.
+    /// Otherwise it will return `None`.
+    ///
+    /// ```
+    /// # use pathbuf::PathComponent;
+    /// # #[cfg(unix)]
+    /// # {
+    /// let some_valid_folder: PathComponent = PathComponent::new("foo").unwrap();
+    /// let some_valid_file: PathComponent = PathComponent::new("bar.txt").unwrap();
+    /// assert!(PathComponent::new("/etc/shadow").is_none());
+    /// # }
+    /// ```
+    pub fn new<S: Into<PathBuf>>(component: S) -> Option<Self> {
+        let component = Self {
+            path: component.into(),
+        };
+
+        component.is_valid().then_some(component)
+    }
+
+    #[cfg(feature = "sanitise")]
+    /// This will sanitise the input and therefore all inputs are valid.
+    /// Unless there is a bug in the sanitisation then it would `panic`.
+    ///
+    /// ```
+    /// # use pathbuf::PathComponent;
+    /// # #[cfg(unix)]
+    /// # {
+    /// assert_eq!(
+    ///     PathComponent::with_sanitise("/etc/shadow"),
+    ///     PathComponent::new("etc_shadow").unwrap(),
+    /// );
+    /// # }
+    /// ```
+    ///
+    /// The sanitisation algorithm isn't considered stable.
+    /// Therefore the sanitised path could change in the future for the same input.
+    pub fn with_sanitise(component: &str) -> Self {
```

Comment:
I think this method ought to be removed. It can panic, and the doc comment specifically says the algorithm for sanitization is considered unstable.

Reply:
`with_sanitise` can only panic if `sanitize-filename-reader-friendly` is broken.

Reply:
Maybe sanitisation should be removed or at least have a warning as OWASP recommends against it.